### PR TITLE
feat(match-list): Implemented new ui element for match-list

### PR DESCRIPTION
**Pull request by lakatosandrei**

### CODE REVIEW

File: packages/match-list/src/answer.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+          id={id}
+          isOver={isOver}
+          empty={isEmpty(title)}
+          isDragging={isDragging}
+          disabled={disabled}
+          type={type}
+        />
+      </div>
+    );
+    const droppable = connectDropTarget ? connectDropTarget(content) : content;
```
Comment (edeustace) [inline packages/match-list/src/answer.js]: always true?

File: packages/match-list/src/answer.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+    return connectDragSource ? connectDragSource(droppable) : droppable;
```
Comment (edeustace) [inline packages/match-list/src/answer.js]: always true?

File: packages/match-list/src/answer.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    border: '1px solid var(--feedback-correct-bg-color, green)'
+  }
+})(Answer);
+
+const answerTarget = {
+  drop(props, monitor) {
+    const draggedItem = monitor.getItem();
+
+    if (draggedItem.instanceId === props.instanceId) {
+      props.placeAnswer(draggedItem.id, props.index);
```
Comment (edeustace) [inline packages/match-list/src/answer.js]: can we stick with conventions - prop callback start with 'on': `onPlaceAnswer`

File: packages/match-list/src/answer-list.jsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            return (
+              <DragAndDropAnswer
+                key={index}
+                index={index}
+                correct={correctnesArray[index]}
+                draggable={!isEmpty(sessionAnswer)}
+                disabled={disabled}
+                instanceId={instanceId}
+                id={sessionAnswer.id}
+                placeAnswer={(id, index) => placeAnswer(id, index)}
```
Comment (edeustace) [inline packages/match-list/src/answer-list.jsx]: `onPlaceAnswer={}`

File: packages/match-list/src/answer-list.jsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    const {
+      classes,
+      disabled,
+      placeAnswer,
+      instanceId,
+      onRemoveAnswer,
+      model
+    } = this.props;
+    const { config } = model;
+    const correctnesArray = this.getCorrectOrIncorrectArray();
```
Comment (edeustace) [inline packages/match-list/src/answer-list.jsx]: typo

File: packages/match-list/src/answer.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+      outcome && classes[outcome]
+    );
+
+    return (
+      <div className={names} dangerouslySetInnerHTML={{ __html: title }} />
+    );
+  }
+});
+
+export class Answer extends React.Component {
```
Comment (edeustace) [inline packages/match-list/src/answer.js]: test?

File: packages/match-list/src/choices-list.jsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import isEmpty from 'lodash/isEmpty';
+import { DragAnswer } from './answer';
+
+export class ChoicesList extends React.Component {
```
[the remark this answers is not in the document]
Comment (edeustace) [reply packages/match-list/src/choices-list.jsx]: test?

File: packages/match-list/src/index.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import ReactDOM from 'react-dom';
+import debug from 'debug';
+import compact from 'lodash/compact';
+import Main from './main';
+
+const log = debug('pie-ui:graph-lines');
+
+export { Main as Component };
+
+export default class MatchList extends HTMLElement {
```
Comment (edeustace) [inline packages/match-list/src/index.js]: test?

**Comment by lakatosandrei** [comment]
@edeustace 
Thanks!

Made all the requested changes btw.

File: packages/match-list/src/answer-list.jsx [hunk 9]
```diff
@@ -12,12 +12,19 @@ export class AnswerList extends React.Component {
...
+    onPlaceAnswer: PropTypes.func.isRequired,
     instanceId: PropTypes.string.isRequired,
     model: PropTypes.object.isRequired,
     prompt: PropTypes.string
   };
 
+  constructor(props) {
+    super(props);
+
+    this.getAnswerFromSession = this.getAnswerFromSession.bind(this);
```
Comment (edeustace) [inline packages/match-list/src/answer-list.jsx]: i think you can remove these and as part of the class definition use arrow functions: 
```javascript
class Foo {
  myBoundFn = () => {
  }
}
```

**Pull request merged**